src/locales/ja/parsers/JPTimeExpressionParser.ts

```diff
@@ -0,0 +1,281 @@
+            return null;
+        }
+
+        const result = context.createParsingResult(match.index, match[0]);
+
+        let hour = 0;
+        let minute = 0;
+        let meridiem = -1;
+
+        // ----- Second
+        if (match[SECOND_GROUP]) {
+            let second = parseInt(toHankaku(match[SECOND_GROUP]));
+            if (isNaN(second)) {
+                second = jaStringToNumber(match[SECOND_GROUP]);
+            }
+            if (second >= 60) return null;
+            result.start.assign("second", second);
+        }
+
+        hour = parseInt(toHankaku(match[HOUR_GROUP]));
+        if (isNaN(hour)) {
+            hour = jaStringToNumber(match[HOUR_GROUP]);
+        }
+
+        // ----- Minutes
+        if (match[MINUTE_GROUP]) {
+            if (match[MINUTE_GROUP] === "半") {
+                minute = 30;
+            } else {
+                minute = parseInt(toHankaku(match[MINUTE_GROUP]));
+                if (isNaN(minute)) {
+                    minute = jaStringToNumber(match[MINUTE_GROUP]);
+                }
+            }
+        } else if (hour > 100) {
+            minute = hour % 100;
+            hour = Math.floor(hour / 100);
+        }
+
+        if (minute >= 60) {
+            return null;
+        }
+
+        if (hour > 24) {
+            return null;
+        }
+        if (hour >= 12) {
+            meridiem = Meridiem.PM;
+        }
+
+        // ----- AM & PM
+        if (match[AM_PM_HOUR_GROUP_1]) {
+            if (hour > 12) return null;
+            const AMPMString = match[AM_PM_HOUR_GROUP_1];
+            const FirstAMPMString = AMPMString[0].toLowerCase();
+            if (AMPMString === "午前" || FirstAMPMString === "a") {
+                meridiem = Meridiem.AM;
+                if (hour === 12) hour = 0;
+            } else if (AMPMString === "午後" || FirstAMPMString === "p") {
+                meridiem = Meridiem.PM;
+                if (hour != 12) hour += 12;
+            }
+        } else if (match[AM_PM_HOUR_GROUP_2]) {
+            if (hour > 12) return null;
+            const ampm = match[AM_PM_HOUR_GROUP_2][0].toLowerCase();
+            if (ampm === "a") {
+                meridiem = Meridiem.AM;
+                if (hour === 12) hour = 0;
+            }
+
+            if (ampm === "p") {
+                meridiem = Meridiem.PM;
+                if (hour != 12) hour += 12;
+            }
+        }
+
+        result.start.assign("hour", hour);
+        result.start.assign("minute", minute);
+
+        if (meridiem >= 0) {
+            result.start.assign("meridiem", meridiem);
+        } else {
+            if (hour < 12) {
+                result.start.imply("meridiem", 0);
+            } else {
+                result.start.imply("meridiem", 1);
+            }
+        }
+
+        // =============================================================================================
+        //                  Extracting the 'to' chunk
```

Comment:
Please avoid full-page line comments like this. Other people (or your IDE vs Github) may have different screen width settings.